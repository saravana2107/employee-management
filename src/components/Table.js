import React from "react";
import { Table } from "reactstrap";

export const TableColumn = () => {};

export default class TableComponent extends React.Component {
  render() {
    const { children: columns, data, notFoundText } = this.props;

    return (
      <Table striped responsive>
        <thead className="thead-light">
          <tr>
            {React.Children.map(columns, x => {
              return x && <th>{x.props.children}</th>;
            })}
          </tr>
        </thead>
        <tbody>
          {data.length === 0 && (
            <tr>
              <td colSpan={columns.length} className="text-center">
                {notFoundText || "No Results Found"}
              </td>
            </tr>
          )}
          {data.map((detail, key) => {
            return (
              <tr key={key}>
                {React.Children.map(columns, (x, index) => {
                  return index === 0 ? (
                    <th scope="row" className="align-middle">
                      {x.props.renderField
                        ? x.props.renderField(detail)
                        : detail[x.props.field]}
                    </th>
                  ) : (
                    <td className="align-middle">
                      {x.props.renderField
                        ? x.props.renderField(detail)
                        : detail[x.props.field]}
                    </td>
                  );
                })}
              </tr>
            );
          })}
        </tbody>
      </Table>
    );
  }
}
