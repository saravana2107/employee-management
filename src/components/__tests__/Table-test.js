import React from "react";
import renderer from "react-test-renderer";
import Table, { TableColumn } from "../Table";

describe("Table", () => {
  it("should display list", () => {
    const data = [
      {
        id: "ALTI123456",
        name: "Test User #1",
        email: "test+1@altimetric.com",
        phone: "9876543210",
        address: "Kudlu Gate",
        city: "Bangalore",
        state: "Karnataka",
        zipcode: "560068",
        gender: "Male"
      }
    ];

    const component = renderer.create(
      <Table data={data} notFoundText="No Employees Found">
        <TableColumn field="id">ID</TableColumn>
        <TableColumn field="name">Name</TableColumn>
        <TableColumn field="email">Email</TableColumn>
        <TableColumn field="phone">Phone</TableColumn>
        <TableColumn field="address">Address</TableColumn>
        <TableColumn field="city">City</TableColumn>
        <TableColumn field="state">State</TableColumn>
        <TableColumn field="zipcode">ZipCode</TableColumn>
        <TableColumn field="gender">Gender</TableColumn>
        <TableColumn
          field="action"
          renderField={row => (
            <div>
              <i className="fas fa-edit" />
              <i className="fas fa-trash" />
            </div>
          )}
        >
          Action
        </TableColumn>
      </Table>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should display not found text if data is empty", () => {
    const component = renderer.create(
      <Table data={[]} notFoundText="No Employees Found">
        <TableColumn field="id">ID</TableColumn>
        <TableColumn field="name">Name</TableColumn>
        <TableColumn field="email">Email</TableColumn>
        <TableColumn field="phone">Phone</TableColumn>
        <TableColumn field="address">Address</TableColumn>
        <TableColumn field="city">City</TableColumn>
        <TableColumn field="state">State</TableColumn>
        <TableColumn field="zipcode">ZipCode</TableColumn>
        <TableColumn field="gender">Gender</TableColumn>
        <TableColumn
          field="action"
          renderField={row => (
            <div>
              <i className="fas fa-edit" />
              <i className="fas fa-trash" />
            </div>
          )}
        >
          Action
        </TableColumn>
      </Table>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should display default not found text if data is empty", () => {
    const component = renderer.create(
      <Table data={[]}>
        <TableColumn field="id">ID</TableColumn>
        <TableColumn field="name">Name</TableColumn>
        <TableColumn field="email">Email</TableColumn>
        <TableColumn field="phone">Phone</TableColumn>
        <TableColumn field="address">Address</TableColumn>
        <TableColumn field="city">City</TableColumn>
        <TableColumn field="state">State</TableColumn>
        <TableColumn field="zipcode">ZipCode</TableColumn>
        <TableColumn field="gender">Gender</TableColumn>
        <TableColumn
          field="action"
          renderField={row => (
            <div>
              <i className="fas fa-edit" />
              <i className="fas fa-trash" />
            </div>
          )}
        >
          Action
        </TableColumn>
      </Table>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
