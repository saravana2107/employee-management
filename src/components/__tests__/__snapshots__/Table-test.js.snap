// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should display default not found text if data is empty 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Phone
        </th>
        <th>
          Address
        </th>
        <th>
          City
        </th>
        <th>
          State
        </th>
        <th>
          ZipCode
        </th>
        <th>
          Gender
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="text-center"
          colSpan={10}
        >
          No Results Found
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table should display list 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Phone
        </th>
        <th>
          Address
        </th>
        <th>
          City
        </th>
        <th>
          State
        </th>
        <th>
          ZipCode
        </th>
        <th>
          Gender
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          className="align-middle"
          scope="row"
        >
          ALTI123456
        </th>
        <td
          className="align-middle"
        >
          Test User #1
        </td>
        <td
          className="align-middle"
        >
          test+1@altimetric.com
        </td>
        <td
          className="align-middle"
        >
          9876543210
        </td>
        <td
          className="align-middle"
        >
          Kudlu Gate
        </td>
        <td
          className="align-middle"
        >
          Bangalore
        </td>
        <td
          className="align-middle"
        >
          Karnataka
        </td>
        <td
          className="align-middle"
        >
          560068
        </td>
        <td
          className="align-middle"
        >
          Male
        </td>
        <td
          className="align-middle"
        >
          <div>
            <i
              className="fas fa-edit"
            />
            <i
              className="fas fa-trash"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table should display not found text if data is empty 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Phone
        </th>
        <th>
          Address
        </th>
        <th>
          City
        </th>
        <th>
          State
        </th>
        <th>
          ZipCode
        </th>
        <th>
          Gender
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="text-center"
          colSpan={10}
        >
          No Employees Found
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
